### ISSUE TITLE
ESLint wants properties that are constructors to use object literal shorthand.

### ISSUE THREAD

**Issue by qubyte** [eslint/eslint#4487]
With the `object-shorthand` rule active, ESLint complains at the following snippet:

```javascript
const test = {
  MyConstructor: function () {}
};
```

It wants it to look like:

```javascript
const test = {
  MyConstructor () {}
};
```

The issue is that object literal shorthands [cannot be used as constructors](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions#Method_definitions_are_not_constructable), so this transformation is invalid.

Seen with eslint@1.9.0.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
If you want to use it as a constructor, use class, not function.

**Comment by qubyte** [comment]
Perhaps, however that doesn't change that the shown ESLint behaviour is in error.

**Comment by qubyte** [comment]
(assuming the convention that constructors use an initial upper cased character)

**Comment by michaelficarra** [comment]
This rule shouldn't care about casing. If you plan to call it as a function, use a method. If you want to construct it, use a class.

**Comment by nzakas** [comment]
We can add an option for this.

**Comment by michaelficarra** [comment]
@nzakas Can you elaborate what you mean by "this"?

**Comment by nzakas** [comment]
Yes, we can add an option that omits uppercase methods from this check.

**Comment by qubyte** [comment]
Thanks folks. Greatly appreciate. :)

**Comment by nrotta** [comment]
I'm currently using the object-shorthand rule to flag the following code as valid (as previously discussed in this thread):
```
const User = BaseModel.extend({
  constructor: function (attrs) {
    objectAssign(this, attrs);
  }
});
```
I'm using eslint **v1.10.3** and the following is the content of my .eslintrc.js file:
```
module.exports = {
  extends: "airbnb",
  rules: {
    "comma-dangle": [2, "never"],
    strict: 0,
    "no-console": 0,    
    "object-shorthand": [2, "always", { "ignoreConstructors": true }]
  }
};
```
However, when running eslint the following error is raised:
```
Configuration for rule "object-shorthand" is invalid:
	Value "2,always,[object Object]" has more items than allowed.
    at validateRuleOptions (node_modules/eslint/lib/config/config-validator.js:102:15)
    at eslint/lib/config/config-validator.js:148:13
    at Array.forEach (native)
    at Object.validate (node_modules/eslint/lib/config/config-validator.js:147:35)
    at Object.load (eslint/lib/config/config-file.js:390:19)
    at loadConfig (eslint/lib/config.js:74:33)
    at getLocalConfig (eslint/lib/config.js:177:23)
    at Config.getConfig (eslint/lib/config.js:272:22)
    at processText (eslint/lib/cli-engine.js:203:27)
    at processFile (eslint/lib/cli-engine.js:270:18)
```
Am I missing anything? I would appreciate your help.

**Comment by ilyavolodin** [comment]
@nrotta `ignoreConstructors` option was added to v2.0.0 so it's not available for 1.10.3 You would need to upgrade to latest to use this option.

**Comment by nrotta** [comment]
Works!! @ilyavolodin thanks for the quick response.
